includes/Classifai/Providers/OpenAI/ChatGPT.php

```diff
@@ -379,6 +407,21 @@ public function setup_fields_sections() {
 			]
 		);
 
+		// Custom prompt for excerpt generation.
+		add_settings_field(
```

Comment:
@dkotter This class is getting bigger I think we should focus on abstracting new code. 

For example, We can add a class for the new field. 
code sample :
```php
<?php
interface SettingField{
	/**
	 * This function should return field id.
	 */
	public function id(): string;

	/**
	 * This function should return field name.
	 */
	public function title(): string;

	/**
	 * This function should return field description.
	 */
	public function description(): string;

	/**
	 * This function should return stored value for setting field.
	 * @return mixed
	 */
	public function get_value();
}

interface Registrable {
	/**
	 * Register a service.
	 */
	public function register(): void;
}

interface Renderable {
	/**
	 * This function should output HTML.
	 */
	public function render(): void;
}

abstract class WordPressSettingField implements SettingField, Registrable, Renderable {
	/**
	 * This function should return option id for setting field.
	 */
	public function get_option_id(): string{
		return $this->id();
	}
}

/**
 * Class ExcerptPromptSettingField
 *
 * This class is used to create excerpt prompt field for OpenAI ChatGPT.
 *
 */
class ExcerptPromptSettingField extends WordPressSettingField{

}
```


cc: @iamdharmesh

Reply:
Thanks for the suggestion @ravinderk and yes I agree that the class getting bigger (it may broken into small parts once we refactor the codebase in a feature-first approach). But adding a new class for the new field looks a bit over-engineered approach to me. However, I am open to discussing this further and we may check on a different approach to keep the class size reasonable. @dkotter what do you think?

Reply:
@iamdharmesh 
We can put all things in one class:
- Necessary data declaration about fields like id, label, etc.
- Register to WordPress setting api
- Rendering logic
- Sanitization logic
   - Accessing field value from request 
- Store to database logic
- Accessing value from database logic
- Misc.


I am happy to discuss about long-term benefits of this abstraction.

Reply:
I agree that this could be done better but that is better handled as a separate task. We have talked about doing a larger refactor (making our settings Feature first instead of Provider first) and doing a refactor on how these settings are registered makes sense as part of that effort.